helix-term/src/commands/typed.rs

```diff
@@ -1527,6 +1527,46 @@ fn tree_sitter_scopes(
     Ok(())
 }
 
+fn tree_sitter_highlight_name(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+
+    let pos = doc.selection(view.id).primary().cursor(text);
+
+    let mut highlight_name: &str = "";
+    if let Some(highlight) =
+        helix_core::syntax::get_highlight_for_node_at_position(doc.syntax(), text, pos)
```

Comment:
I don't think we want this function to live in helix-core since I don't think we'll re-use it in the future. Instead let's have a helper function within this function for finding the highlight at the cursor. That will also help us handle the `Option`s a little cleaner via the `?` operator:

```rust
fn find_highlight_at_cursor(
    cx: &mut compositor::Context<'_>,
) -> Option<helix_core::syntax::Highlight> {
    use helix_core::syntax::HighlightEvent;

    let (view, doc) = current!(cx.editor);
    let syntax = doc.syntax()?;
    let text = doc.text().slice(..);
    let cursor = doc.selection(view.id).primary().cursor(text);
    let byte = text.char_to_byte(cursor);
    let node = syntax
        .tree()
        .root_node()
        .descendant_for_byte_range(byte, byte)?;
    // Query the same range as the one used in syntax highlighting.
    let range = {
        // Calculate viewport byte ranges:
        let row = text.char_to_line(view.offset.anchor.min(text.len_chars()));
        // Saturating subs to make it inclusive zero indexing.
        let last_line = text.len_lines().saturating_sub(1);
        let height = view.inner_area(doc).height;
        let last_visible_line = (row + height as usize).saturating_sub(1).min(last_line);
        let start = text.line_to_byte(row.min(last_line));
        let end = text.line_to_byte(last_visible_line + 1);

        start..end
    };

    let mut highlight = None;

    for event in syntax.highlight_iter(text, Some(range), None) {
        match event.unwrap() {
            HighlightEvent::Source { start, end }
                if start == node.start_byte() && end == node.end_byte() =>
            {
                return highlight;
            }
            HighlightEvent::HighlightStart(hl) => {
                highlight = Some(hl);
            }
            _ => (),
        }
    }

    None
}
```

Then we can call that below and return early if there isn't a highlight for the cursor:

```rust
if event != PromptEvent::Validate {
    return Ok(());
}

let Some(highlight) = find_highlight_at_cursor(cx) else {
    return Ok(());
};
```